packages/playwright-core/src/client/network.ts

```diff
@@ -703,7 +701,7 @@ export class RouteHandler {
     // Note that context.route handler may be later invoked on a different page,
     // so we only swallow errors for the current page's routes.
     const handlerActivations = page ? this._activeInvocations.get(page) : [...this._activeInvocations.values()];
-    if (this.noWaitOnUnrouteOrClose || noWait) {
+    if (noWait) {
       handlerActivations.forEach(h => h.ignoreException = true);
     } else {
       const promises = [];
```

Comment:
Do we still need the logic below about `route._didThrow`?

Reply:
I believe we do as we should not be waiting for the handlers that threw an error already.